pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -2086,14 +2100,64 @@ impl Timeline {
         }
         // end: move the above part to the loop body
 
+        let mut rewrote_delta_layers = Vec::new();
+        if let Some(compaction_key_range) = &job_desc.partial_key_range {
+            for layer in job_desc.rewrite_layers {
+                // For each of the layer to rewrite, generate a layer with the same LSN but less key content
+                // This download should be no-op b/c it's included in selected_layer and has been downloaded before
+                let resident_layer = layer.download_and_keep_resident().await?;
```

Comment:
Why not store `ResidentLayer`s in `rewrite_layers` and `selected_layers` then?

Reply:
after the refactor the field won't be read any more, it's now part of the job desc for debugging purpose, so I changed it to LayerDesc instead